### PR TITLE
feat(issuer): [SFEQS-1439] Add a Azure Function to Validate documents

### PR DESCRIPTION
**Pull request by lucacavallaro**
#### List of Changes

1. Add the new function `GetDocumentValidation`

#### Motivation and Context

This feature allows Issuers to validate documents before creating Signature Requests, it is useful in the CTA context to verify documents before blocking signers

#### How Has This Been Tested?

It was e2e tested manually, with valid and malformed documents

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/io-func-sign-issuer/openapi.yaml [hunk 1]
```diff
@@ -682,3 +717,21 @@ components:
...
+          format: binary
+      required:
+        - document_metadata
+        - document
+
+    DocumentValidation:
+      type: object
+      properties:
+        is_valid:
+          type: boolean
```
Comment (grausof) [inline apps/io-func-sign-issuer/openapi.yaml:737]:
```suggestion
    DocumentValidation:
      type: object
      properties:
        is_valid:
          type: boolean
      required:
        - is_valid
```

File: apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      const parts = boundary
+        ? multipart.Parse(Buffer.from(req.body as string), boundary)
+        : [];
+      return parts.reduce(
+        (obj, part) => {
+          switch (part.type) {
+            case "application/json":
+              return {
+                ...obj,
+                document_metadata: JSON.parse(part.data.toString()),
```
Comment (gunzip) [inline apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts:37]: what if this throws? (maybe we can use some json module in fp-ts?)

File: apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+const makeGetDocumentValidationHandler = () => {
+  const requireIsPdfDocumentCompatibleWithMetadataPayload = flow(
+    azure.fromHttpRequest,
+    TE.fromEither,
+    TE.chainIOK((req) => () => {
+      const [, boundary] = req.headers["content-type"]
+        ? req.headers["content-type"].split(/;\sboundary=/i)
+        : [];
+      const parts = boundary
+        ? multipart.Parse(Buffer.from(req.body as string), boundary)
```
Comment (gunzip) [inline apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts:29]: does this throws?

File: apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+import { success, error } from "@io-sign/io-sign/infra/http/response";
+import { HttpBadRequestError } from "@io-sign/io-sign/infra/http/errors";
+import { DocumentMetadata as DocumentMetadataApiModel } from "../../http/models/DocumentMetadata";
+import { DocumentMetadataFromApiModel } from "../../http/decoders/document";
+
+import { DocumentValidation } from "../../http/models/DocumentValidation";
+import { IsPdfDocumentCompatibleWithMetadata } from "../../../app/use-cases/get-document-validation";
+
+const makeGetDocumentValidationHandler = () => {
```
Comment (gunzip) [inline apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts:20]: needs unit tests

File: apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  const requireIsPdfDocumentCompatibleWithMetadataPayload = flow(
+    azure.fromHttpRequest,
+    TE.fromEither,
+    TE.chainIOK((req) => () => {
+      const [, boundary] = req.headers["content-type"]
+        ? req.headers["content-type"].split(/;\sboundary=/i)
+        : [];
+      const parts = boundary
+        ? multipart.Parse(Buffer.from(req.body as string), boundary)
+        : [];
```
Comment (grausof) [inline apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts:30]: could you integrate this stuff on handler-kit?

**Review by grausof — APPROVED** [review]

File: apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts (hunk 5 above)
Comment (lucacavallaro) [reply apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts:30]: I am releasing this week a new breaking version of `handler-kit`, for this reason I decided to integrate these functions here for now, I will no longer update the current package

File: apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts (hunk 3 above)
Comment (lucacavallaro) [reply apps/io-func-sign-issuer/src/infra/azure/functions/get-document-validation.ts:29]: nope, this at least produces "unexpected" arrays that are sanitized later

**Pull request closed without merging**